pkg/pipeline/engine/jenkins/mappper.go

```diff
@@ -215,6 +238,61 @@ func (c *jenkinsPipelineConverter) injectGitCaCert(pod *v1.Pod) {
 	})
 }
 
+func getImagePullSecretNames(secretLister apiv1.SecretLister, execution *v3.PipelineExecution) ([]string, error) {
+	result := []string{}
+	ns := utils.GetPipelineCommonName(execution.Spec.ProjectName)
+	availableCreds, err := getDomainToCreds(secretLister, ns)
+	if err != nil {
+		return nil, err
+	}
+	for _, stage := range execution.Spec.PipelineConfig.Stages {
+		for _, step := range stage.Steps {
+			if step.RunScriptConfig != nil {
+				domain := getDomain(step.RunScriptConfig.Image)
+				result = append(result, availableCreds[domain]...)
+			}
+		}
+	}
+	logrus.Debugf("using imagepullsecrets %v for the build", result)
+	return result, nil
+}
+
+func getDomainToCreds(secretLister apiv1.SecretLister, namespaceID string) (map[string][]string, error) {
```

Comment:
Can we just add all the image pull secret names to pods instead of validating the domain and json content? Or do I miss anything?

Reply:
we can do that, updated